### ISSUE TITLE
Plugin doesn't compile my scss files when upgraded to 1.2.5

### ISSUE THREAD

**Issue by RawMajkel** [ConnectThink/WP-SCSS#131]
Literally after upgarding to 1.2.5 compiling doesn't work anymore - any changes made in .scss files are not seen in compiled css nor on the result web page. I had to make myself an 1.2.4 version of plugin in a .zip file to keep it.

I do have my config looking like this: https://bit.ly/2sZCOdP

My style.scss file is full of "@import"s but it should not be a problem especially when everything worked great one version before.

Anything I could try to make it work? At this point I just don't update this plugin

**Comment by johntbb** [comment]
It's this commit that's stopped changes in imported files from triggering complication:

https://github.com/ConnectThink/WP-SCSS/pull/111/commits/12e0108d18d0abe5df2b7c2fe706b291fc5671e7

I basically reverted this commit manually in my code, and everything has started working again and I don't need to keep re-saving the main scss file for compilation to trigger. I guess (or at least hope) that the next version will fix this issue, so I don't need to do this again.

**Comment by johntbb** [comment]
Going one better than my last comment, I actually fixed the code:

`new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->css_dir, RecursiveDirectoryIterator::SKIP_DOTS))` instead of `new RecursiveIteratorIterator(new RecursiveDirectoryIterator($this->scss_dir), FilesystemIterator::SKIP_DOTS)`

**Comment by RawMajkel** [comment]
@johntbb That's actually pretty cool, thanks. I'd be nice to see this getting fixed in next version of the plugin, because I don't like to leave my plugins outdated.

**Comment by garrettw** [comment]
Just for documentation purposes... Duplicate of #130

**Comment by RawMajkel** [comment]
@johntbb Finally I've reailsed what my problem with WordPress was. I've noticed lately that WordPress loading time is up to 5 seconds 😳 (btw I'm even using webp images to improve my performance). So after loading times like 500ms I've got punched with 5s suddenly. I was getting really nervous because I didn't know what was the probley. Finally today I've noticed that it was just WP-SCSS 1.2.5 with that "community fix" you've proposed.

So basically I really recommend staying with 1.2.4 for a while and not changing anything in plugin code. Maybe that will help someone.

**Comment by johntbb** [comment]
> @johntbb Finally I've reailsed what my problem with WordPress was. I've noticed lately that WordPress loading time is up to 5 seconds 😳 (btw I'm even using webp images to improve my performance). So after loading times like 500ms I've got punched with 5s suddenly. I was getting really nervous because I didn't know what was the probley. Finally today I've noticed that it was just WP-SCSS 1.2.5 with that "community fix" you've proposed.
> 
> So basically I really recommend staying with 1.2.4 for a while and not changing anything in plugin code. Maybe that will help someone.

I'm surprised. I haven't noticed any slow downs on any of my projects when using this fix.

**Comment by agkovalev** [comment]
Thanks, it works!

**Comment by anderslundyo** [comment]
This is still a problem for me. 

Im targeting a stylesheet only with imports, and saving into these doesnt trigger the compiler. I need to save the targeted file. It's a 1.2.5 thing, and it needs fixing now.

**Comment by RawMajkel** [comment]
@anderslundyo It is true. Therefore I've given up on this plugin. I strongly recommend using scssphp (less plugins installed, always working), or to just wait for developer to actually do something about this bug.

**Comment by dustyyy** [comment]
A year after @johntbb proposed a fix and this is still an issue for me.

**Comment by johntbb** [comment]
> A year after @johntbb proposed a fix and this is still an issue for me.

There's another scss plugin available which is maintained now. I would not recommend using this one anymore as several of the libraries it uses are out of date.

**Comment by shadoath** [comment]
Taking over the maintenance of this plugin. The creators did not have time to work on it. Now, i'm looking at the backlog of issues and working to solve them. @johntbb, thank you for pointing out the solution. I have a [PR](https://github.com/ConnectThink/WP-SCSS/pull/166) with updates you can look at.

**Comment by shadoath** [comment]
Duplicate of #130
